lib/puppet/configurer.rb

```diff
@@ -410,7 +410,13 @@ def send_report(report)
   end
 
   def save_last_run_summary(report)
-    mode = Puppet.settings.setting(:lastrunfile).mode
+    # Warning - this is ugly - if anyone can suggest a better way to do this and get the
+    # corresponding spec tests to work ok - please do.
+    chmode = Puppet.settings.setting(:lastrunfile).mode
+    mode = chmode.to_i(8)
```

Comment:
`Integer(mode, 8)` will raise an argument error if you don't pass a numeric value

Reply:
Done 💯 suggestion.